sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOAutoPartitioningIT.java

```diff
@@ -0,0 +1,416 @@
+
+  public static final Integer NUM_ROWS = 1_000;
+  public static final String TABLE_NAME = "baseTable";
+
+  @ClassRule public static TestPipeline pipelineWrite = TestPipeline.create();
+  @Rule public TestPipeline pipelineRead = TestPipeline.create();
+
+  @Parameterized.Parameters(name = "{0}")
+  public static Iterable<String> params() {
+    return Lists.newArrayList("mysql", "postgres");
+  }
+
+  @Parameterized.Parameter(0)
+  public String dbms;
+
+  public static JdbcDatabaseContainer<?> getDb(String dbName) {
+    if (dbName.equals("mysql")) {
+      return mysql;
+    } else {
+      return postgres;
+    }
+  }
+
+  // We need to implement this retrying rule because we're running ~18 pipelines that connect to
+  // two databases in a few seconds. This may cause the databases to be overwhelmed, and reject
+  // connections or error out. By using this rule, we ensure that each pipeline is trued twice so
+  // that flakiness from databases being overwhelmed can be managed.
+  @Rule
+  public TestRule retryRule =
+      new TestRule() {
+        // We establish max number of retries at 2.
+        public final int maxRetries = 2;
+
+        @Override
+        public Statement apply(Statement base, Description description) {
+          return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+              Throwable caughtThrowable = null;
+              // implement retry logic here
+              for (int i = 0; i < maxRetries; i++) {
+                try {
+                  pipelineRead.apply(base, description);
+                  base.evaluate();
+                  return;
+                } catch (Throwable t) {
+                  caughtThrowable = t;
+                  System.err.println(
+                      description.getDisplayName() + ": run " + (i + 1) + " failed.");
+                }
+              }
+              System.err.println(
+                  description.getDisplayName() + ": Giving up after " + maxRetries + " failures.");
+              throw Objects.requireNonNull(caughtThrowable);
+            }
+          };
+        }
+      };
+
+  @ClassRule public static JdbcDatabaseContainer<?> mysql = new MySQLContainer<>("mysql");
+
+  @ClassRule
+  public static JdbcDatabaseContainer<?> postgres = new PostgreSQLContainer<>("postgres");
+
+  @Before
+  public void prepareDatabase() throws SQLException {
+    pipelineRead.getOptions().setStableUniqueNames(CheckEnabled.OFF);
+    DataSource dbDs = DatabaseTestHelper.getDataSourceForContainer(getDb(dbms));
+    try {
+      DatabaseTestHelper.createTable(
+          dbDs,
+          TABLE_NAME,
+          Lists.newArrayList(
+              KV.of("id", "INTEGER"),
+              KV.of("name", "VARCHAR(50)"),
+              KV.of("specialDate", "TIMESTAMP")));
+    } catch (SQLException e) {
+      LOG.warn("Exception occurred when preparing database {}", dbms, e);
```

Comment:
Logging a warning here might be confusion, maybe just comment that this is expected as the db already exists when running this before the 2nd test...? Though, nothing critical ....